distribution/sender/routes/debezium-route.xml

```diff
@@ -3,88 +3,101 @@
-                #{getProperty('event').setOperation(request.headers.CamelDebeziumOperation)}
-                #{getProperty('event').setSnapshot(request.headers.CamelDebeziumSourceMetadata.get('snapshot') != 'false')}
+                #{getProperty('debeziumEvent').event.setPrimaryKeyId(request.headers.CamelDebeziumKey.get(request.headers.CamelDebeziumKey.schema().fields().get(0)))}
+                #{getProperty('debeziumEvent').event.setTableName(request.headers.CamelDebeziumSourceMetadata.get('table'))}
+                #{getProperty('debeziumEvent').event.setOperation(request.headers.CamelDebeziumOperation)}
+                #{getProperty('debeziumEvent').event.setSnapshot(request.headers.CamelDebeziumSourceMetadata.get('snapshot') != 'false')}
+                #{getProperty('debeziumEvent').setDateCreated(new java.util.Date())}
+      			#{getProperty('debeziumEvent').setEventTimeStamp(request.headers.CamelDebeziumTimestamp)}
             </spel>
         </script>
+
         <setProperty name="is-subclass">
-            <spel>#{getProperty('event').tableName == 'test_order' || getProperty('event').tableName == 'drug_order' || getProperty('event').tableName == 'patient'}</spel>
+            <spel>#{getProperty('debeziumEvent').event.tableName == 'test_order' || getProperty('debeziumEvent').event.tableName == 'drug_order' || getProperty('debeziumEvent').event.tableName == 'patient'}</spel>
         </setProperty>
 
         <choice>
             <when>
                 <simple>${headers.CamelDebeziumOperation} == 'c' || ${headers.CamelDebeziumOperation} == 'u' || ${headers.CamelDebeziumOperation} == 'd'</simple>
-                <when>
+              
+             	 <when>
                     <simple>${exchangeProperty.is-subclass} == false</simple>
                     <choice>
                         <when>
                             <simple>${headers.CamelDebeziumOperation} == 'd'</simple>
                             <script>
-                                <spel>#{getProperty('event').setIdentifier(request.headers.CamelDebeziumBefore.get('uuid'))}</spel>
+                                <spel>#{getProperty('debeziumEvent').event.setIdentifier(request.headers.CamelDebeziumBefore.get('uuid'))}</spel>
                             </script>
                         </when>
                         <otherwise>
                             <script>
-                                <spel>#{getProperty('event').setIdentifier(body.get('uuid'))}</spel>
+                                <spel>#{getProperty('debeziumEvent').event.setIdentifier(body.get('uuid'))}</spel>
                             </script>
                         </otherwise>
                     </choice>
                 </when>
+                <otherwise>
+		            <!-- For a subclass table, fetch the identifier(uuid) from the parent table e.g person.uuid value for patient table joining on the FK -->
+		            <choice>
```

Comment:
This logic should not be in this route

Reply:
I already commented that this should be here. I can't figure out how will you find the UUID after the record have been removed. Unless you mean to create a new route and put this logic and then call this new route from debezium-route